src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -515,6 +524,20 @@ private void CompileNamedType(NamedTypeSymbol containingType)
                             {
                                 hasStaticConstructor = true;
                             }
+
+                            if (_moduleBeingBuiltOpt is object && method is SourceMethodSymbolWithAttributes { MethodKind: MethodKind.Ordinary, IsModuleInitializer: true })
```

Comment:
Is this the right place for this code? Inside CompileMethod didn't sound right by the name and this attribute wouldn't apply to the other three places where CompileMethod is called.